tests/test_openslide.py

```diff
@@ -44,7 +59,11 @@ def test_open(self):
         self.assertRaises(
             OpenSlideUnsupportedFormatError, lambda: OpenSlide('setup.py')
         )
-        self.assertRaises(OpenSlideError, lambda: OpenSlide('unopenable.tiff'))
+        self.assertRaises(OpenSlideUnsupportedFormatError, lambda: OpenSlide(None))
+        self.assertRaises(OpenSlideUnsupportedFormatError, lambda: OpenSlide(3))
```

Comment:
Interesting, some functions return ArgumentError, others return TypeError, but I wasn't expecting OpenSlideUnsupportedFormatError.

But then again, I don't think any of the code in this pull request is related to this confusion, so you can ignore this comment :)

Reply:
Yeah, it's not ideal, but it's what the code currently does.  I opted to add a test for the current behavior rather than potentially breaking compatibility by changing it.

Reply:
UnsupportedFormatError could inherit from ArgumentError, making it less likely to break compatibility because the old check would still work. But we can't deprecate unsupportedformat so the end result would be forcing all users to test for multiple exceptions instead of just one which doesn't seem like much of an improvement.